src/jit/codegenarm64.cpp

```diff
@@ -1558,6 +1558,24 @@ void CodeGen::genCodeForBinary(GenTree* treeNode)
     GenTreePtr  op2 = treeNode->gtGetOp2();
     instruction ins = genGetInsForOper(treeNode->OperGet(), targetType);
 
+    if ((treeNode->gtFlags & GTF_SET_FLAGS) != 0)
+    {
+        switch (oper)
+        {
+            case GT_ADD:
+                ins = INS_adds;
+                break;
+            case GT_SUB:
+                ins = INS_subs;
+                break;
+            case GT_AND:
+                ins = INS_ands;
+                break;
+            default:
+                assert(!"Unexpected BinaryOp with GTF_SET_FLAGS set");
```

Comment:
Shouldn't this be a noway_assert instead?

Reply:
@briansull I would have to defer to your judgment.  I assume noway_assert exists in release mode too.

Reply:
Yes, these are "release" asserts, where as a vanilla assert is a checked only assert.  
If you hit a noway_assert, when running a release build :
1) We can get telemetry indicating that it was hit
2) And the VM layer catches the noway_assert,and will retry the JIT compilation once with optimization disabled.